packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -713,6 +731,45 @@ Map<String, String> parseAdbDeviceProperties(String str) {
   return properties;
 }
 
+
+const String _kJavaHeapKey = 'Java Heap';
+const String _kNativeHeapKey = 'Native Heap';
+const String _kGraphicsKey = 'Graphics';
+
+/// Process the dumpsys info formatted in a table-like structure.
+///
+/// Example output:
+///
+/// Applications Memory Usage (in Kilobytes):
+/// Uptime: 441088659 Realtime: 521464097
+///
+/// ** MEMINFO in pid 16141 [io.flutter.demo.gallery] **
+///                   Pss  Private  Private  SwapPss     Heap     Heap     Heap
+///                 Total    Dirty    Clean    Dirty     Size    Alloc     Free
+///                ------   ------   ------   ------   ------   ------   ------
+/// Native Heap      8648     8620        0       16    20480    12403     8076
+/// Dalvik Heap       547      424       40       18     2628     1092     1536
+/// Dalvik Other      464      464        0        0
+/// Stack             496      496        0        0
+/// ...
+///
+/// For more information, see https://developer.android.com/studio/command-line/dumpsys.
+@visibleForTesting
+Map<String, Object> parseMeminfoDump(String input) {
+  final Map<String, Object> result = <String, Object>{};
+  for (String line in input.split('\n')) {
+    line = line.trim();
+    if (line.startsWith(_kJavaHeapKey)) {
+      result[_kJavaHeapKey] = int.tryParse(line.split(':').last.trim());
```

Comment:
There are no colons in the example above or in the linked documentation.

Reply:
Parsing of 'TOTAL:' would be a little different because of 'TOTAL SWAP PSS:' on the same line I would ignore 'TOTAL:' as the other 7 will add up to this anyway.